src/azul/deployment.py

```diff
@@ -297,5 +301,71 @@ def _sanitize_tf(tf_config: JSON) -> JSON:
+    >>> tags = fixed['resource'][0]['aws_resource_type']['resource_name']['tags']
+    >>> assert 'Name' in tags
+    >>> assert 'name' not in tags
+    """
+    for group in filter(lambda x: isinstance(x, dict), tf_config.get('resource', [])):
+        for resource_name, resource_data in group.items():
+            if resource_name.startswith('aws_'):
+                for service in resource_data.values():
+                    if 'tags' in service and 'name' in service['tags']:
+                        service['tags']['Name'] = service['tags'].pop('name')
+    return tf_config
+
+
 def emit_tf(tf_config: Optional[JSON]):
-    return emit(tf_config) if tf_config is None else emit(_sanitize_tf(tf_config))
+    return emit(tf_config) if tf_config is None else emit(_fix_name_tag(_sanitize_tf(tf_config)))
+
+
+def get_cloud_tags(resource_name: str, **kwargs) -> dict:
+    """
+    Returns tags named for cloud resources based on :class:`azul.Config`.
+    If no ``resource_name`` is passed, a kwarg ``name`` is expected.
+
+    The following tags are returned:
+    * ``name``, which is provided by ``resource_name``. It should be set to a
+      unique fully-qualified resource name (e.g. ``azul-foo-dev``) unless the
+      resource being tagged already has a name assigned to it, in which case
+      that name should be used, even if it is not unique. The unique
+      fully-qualified resource name can be generated using
+      :meth:`azul.Config.qualified_resource_name`.
+    * ``project``, set for now to ``dcp``.
+    * ``service``, set to $AZUL_RESOURCE_PREFIX.
+    * ``deployment``, set to $AZUL_DEPLOYMENT_STAGE.
+    * ``owner``, set to $AZUL_OWNER.
+    * ``creator``, set to the ``UserId`` value from a GetCallerIdentity call to
+      the AWS Security Token Service.
+    * ``component``, which is generally the ``name`` tag without the trailing
+      stage to allow resources to be filtered across stages. (Using the example
+      above, the ``component`` value for ``azul-foo-dev`` would be ``azul-foo``.)
+      In cases where the ``name`` tag does not take the form of a fully-qualified
+      resource name as above, the value of this tag should be provided explicitly
+      by the ``component`` kwarg.
+
+    :param str resource_name: Value for the ``name`` tag of the resource
+    :param kwargs: Provided key-value pairs can be used to specify
+        additional tags or to override default tags.
+    """
+    if resource_name.endswith("-" + config.deployment_stage):
+        stripped_name = resource_name[:-len(config.deployment_stage) - 1]
+    else:
+        stripped_name = resource_name
+    component_suffix = config.unqualified_resource_name_or_none(resource_name)[0] or stripped_name
+    return {
+        'project': 'dcp',
+        'service': config._resource_prefix,
+        'deployment': config.deployment_stage,
+        'owner': config.owner,
+        'creator': aws.user,
```

Comment:
I was unable to make progress with this, and have removed the `creator` tag entirely. I did find the [`ignore_tags`](https://www.terraform.io/docs/providers/aws/index.html#ignore_tags-configuration-block) property, but it didn't appear to work in the fashion that was required.

Reply:
I linked to a post about `ignore_changes` but you mention to `ignore_tags`. Did you try the former?

Reply:
Yes -- I took some time today to verify again. I tested with Terraform v0.12.24:

* Using only `ignore_tags` (at the provider level)
* Using only `ignore_changes` (for each resource)
* Using both

None of these approaches were effective; in all of my tests, none of these methods were sufficient to prevent Terraform from trying to update tags on resources that otherwise did not need to be modified.

There is a possibility that I'm doing something wrong -- per the issue that you linked, some people have been able to do this successfully as recently as last year. That being said, there are also a number of people who are unable to reproduce those successes, even though the issue was marked as resolved (hashicorp/terraform#21788).

In any case, it might be worth revisiting this down the line. There is an open Terraform feature request (hashicorp/terraform#20866) that would make implementing this functionality as described much easier.

Reply:
I'd like to see the version of this PR that uses the technique I linked to and a terminal transcript that proves that it doesn't work.

Reply:
I've tagged you on Slack with a link to a Gist containing relevant transcripts.

Reply:
@hannes-ucsc and I have discussed this at length. The `creator` tag is being removed as the behavior we need (i.e., configuring a tag such that it is only updated if another part of the resource is updated) is not readily achievable. Terraform allows ignoring changes to a tag and updating it only on resource creation or replacement, but this behavior is insufficient for this use case.